File: src/modules/user/services/user_update_status_service.py
from django.db.models import Q
from src.shared.types.user_request import UserRequest
from src.modules.user.models import User
from src.modules.user.domain.dto.user_update_status_dto import (
    UserUpdateStatusDTO,
)
from src.shared.exceptions.not_found_exception import NotFoundException


class UserUpdateStatusService:
    @classmethod
    def execute(self, id: int, dto: UserUpdateStatusDTO, user_request: UserRequest):

        user = User.objects.filter(
            Q(id=id)
            & Q(deleted_at=None)
            & (
                Q(organization__id=user_request.organization.id)
                if user_request.organization
                else Q()
            )
            & (Q(company__id=user_request.company.id) if user_request.company else Q())
        ).first()

        if user is None:
            raise NotFoundException("Usuário não encontrado!")

        user.status = dto.status

        user.save()

        return user

File: src/modules/user/services/user-update-status.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';
import { UserRequest } from 'src/shared/types/user-request';
import { UserUpdateStatusDto } from '../domain/dto/user-update-status.dto';
import { UserEntity } from '../domain/entities/user.entity';

@Injectable()
export class UserUpdateStatusService {
  constructor(private prisma: PrismaService) {}
  async execute(
    id: number,
    dto: UserUpdateStatusDto,
    userRequest: UserRequest,
  ): Promise<UserEntity> {
    const user = await this.prisma.user.findFirst({
      where: {
        id: id,
        organizationId: userRequest?.organizationId ?? undefined,
        companyId: userRequest?.companyId ?? undefined,
        deletedAt: null,
      },
    });

    if (!user) throw new NotFoundException('Usuário não encontrado!');

    const updatedUser = await this.prisma.user.update({
      data: {
        status: dto.status,
      },
      where: {
        id: id,
      },
      include: {
        role: true,
        organization: true,
        company: true,
        avatar: true,
      },
    });

    return new UserEntity(updatedUser);
  }
}
